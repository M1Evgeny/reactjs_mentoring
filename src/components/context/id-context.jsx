import React from 'react';

const IdContext = React.createContext();

export function IdProvider(props) {
  const [movieId, setMovieId] = React.useState(0);
  const value = React.useMemo(() => ({ movieId, setMovieId }), [movieId]);
  return <IdContext.Provider value={value} {...props} />
}

export function useId() {
  const context = React.useContext(IdContext);
  if (!context) {
    throw new Error(`useId must be used within a IdProvider`);
  }
  return context;
}
