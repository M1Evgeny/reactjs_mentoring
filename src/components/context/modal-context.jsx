import React from 'react';

const ModalContext = React.createContext();

export function ModalProvider(props) {
    const [modalObject, setModalObject] = React.useState({});
    const value = React.useMemo(() => ({ modalObject, setModalObject }), [modalObject]);
    return <ModalContext.Provider value={value} {...props} />;
}

export function useModal() {
    const context = React.useContext(ModalContext);
    if (!context) {
      throw new Error(`useModal must be used within a ModalContext`);
    }
    return context; 
}
