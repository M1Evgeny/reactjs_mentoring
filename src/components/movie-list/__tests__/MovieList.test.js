import React from "react";
import { render } from "@testing-library/react";
import { MovieList } from "..";
import stubs from "../../movie-list-container/mockedMovies.json";
import { ModalProvider } from "../../context/modal-context";

describe("MovieList test", () => {
  test("MovieList snapshot", () => {
    const { asFragment } = render(
      <ModalProvider>
        <MovieList movies={stubs} />
      </ModalProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
