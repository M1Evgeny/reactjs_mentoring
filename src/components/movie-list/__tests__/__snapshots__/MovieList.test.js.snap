// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList test MovieList snapshot 1`] = `
<DocumentFragment>
  <div
    class="movie-list"
  >
    <article
      class="filmCard movie-card"
    >
      <div
        class="hamburger_menu"
      >
        <input
          class="menu__toggle"
          id="menu__toggle337167"
          type="checkbox"
        />
        <label
          class="menu__btn"
          for="menu__toggle337167"
        >
          <span />
        </label>
        <ul
          class="menu__box"
        >
          <li>
            <div
              class="menu__item"
              href="#"
              id="edit-button"
            >
              Edit
            </div>
          </li>
          <li>
            <div
              class="menu__item"
              href="#"
              id="delete-button"
            >
              Delete
            </div>
          </li>
        </ul>
      </div>
      <img
        alt="..."
        class="card_img_top"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
      <div
        class="card_body"
      >
        <span
          class="filmTitle movie-card-title"
        >
          Fifty Shades Freed
        </span>
        <span
          class="filmYear"
        >
          2018-02-07
        </span>
        <p
          class="card-text"
        >
          Drama, Romance
        </p>
      </div>
    </article>
    <article
      class="filmCard movie-card"
    >
      <div
        class="hamburger_menu"
      >
        <input
          class="menu__toggle"
          id="menu__toggle338970"
          type="checkbox"
        />
        <label
          class="menu__btn"
          for="menu__toggle338970"
        >
          <span />
        </label>
        <ul
          class="menu__box"
        >
          <li>
            <div
              class="menu__item"
              href="#"
              id="edit-button"
            >
              Edit
            </div>
          </li>
          <li>
            <div
              class="menu__item"
              href="#"
              id="delete-button"
            >
              Delete
            </div>
          </li>
        </ul>
      </div>
      <img
        alt="..."
        class="card_img_top"
        src="https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg"
      />
      <div
        class="card_body"
      >
        <span
          class="filmTitle movie-card-title"
        >
          Tomb Raider
        </span>
        <span
          class="filmYear"
        >
          2018-03-08
        </span>
        <p
          class="card-text"
        >
          Action, Adventure
        </p>
      </div>
    </article>
    <article
      class="filmCard movie-card"
    >
      <div
        class="hamburger_menu"
      >
        <input
          class="menu__toggle"
          id="menu__toggle181808"
          type="checkbox"
        />
        <label
          class="menu__btn"
          for="menu__toggle181808"
        >
          <span />
        </label>
        <ul
          class="menu__box"
        >
          <li>
            <div
              class="menu__item"
              href="#"
              id="edit-button"
            >
              Edit
            </div>
          </li>
          <li>
            <div
              class="menu__item"
              href="#"
              id="delete-button"
            >
              Delete
            </div>
          </li>
        </ul>
      </div>
      <img
        alt="..."
        class="card_img_top"
        src="https://image.tmdb.org/t/p/w500/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg"
      />
      <div
        class="card_body"
      >
        <span
          class="filmTitle movie-card-title"
        >
          Star Wars: The Last Jedi
        </span>
        <span
          class="filmYear"
        >
          2017-12-13
        </span>
        <p
          class="card-text"
        >
          Fantasy, Adventure, Science Fiction
        </p>
      </div>
    </article>
    <article
      class="filmCard movie-card"
    >
      <div
        class="hamburger_menu"
      >
        <input
          class="menu__toggle"
          id="menu__toggle284054"
          type="checkbox"
        />
        <label
          class="menu__btn"
          for="menu__toggle284054"
        >
          <span />
        </label>
        <ul
          class="menu__box"
        >
          <li>
            <div
              class="menu__item"
              href="#"
              id="edit-button"
            >
              Edit
            </div>
          </li>
          <li>
            <div
              class="menu__item"
              href="#"
              id="delete-button"
            >
              Delete
            </div>
          </li>
        </ul>
      </div>
      <img
        alt="..."
        class="card_img_top"
        src="https://image.tmdb.org/t/p/w500/uxzzxijgPIY7slzFvMotPv8wjKA.jpg"
      />
      <div
        class="card_body"
      >
        <span
          class="filmTitle movie-card-title"
        >
          Black Panther
        </span>
        <span
          class="filmYear"
        >
          2018-02-13
        </span>
        <p
          class="card-text"
        >
          Action, Adventure, Fantasy, Science Fiction
        </p>
      </div>
    </article>
  </div>
</DocumentFragment>
`;
